app/src/main/java/org/wikipedia/descriptions/DescriptionEditView.java

```diff
@@ -371,6 +373,18 @@ void validateText() {
         updateSaveButtonEnabled();
     }
 
+    private void checkKeyboardLanguage(String text) {
+        FirebaseLanguageIdentification languageIdentifier =
+                FirebaseNaturalLanguage.getInstance().getLanguageIdentification();
+        languageIdentifier.identifyLanguage(text)
+                .addOnSuccessListener(
+                        languageCode -> {
```

Comment:
Is it possible for this `successListener` to be delayed significantly? (hint: until after this view is disposed?)

Reply:
Now that the view implements a callback this won't be a problem @dbrant